src/utils.ts

```diff
@@ -10,14 +10,56 @@ import {execSync} from 'child_process';
 import * as mkdirp from 'mkdirp';
 const isWsl = require('is-wsl');
 
-export function defaults<T>(val: T | undefined, def: T): T {
+export const enum LaunchErrorCodes {
+  ERR_LAUNCHER_PATH_NOT_SET = 'ERR_LAUNCHER_PATH_NOT_SET',
+  ERR_LAUNCHER_INVALID_USER_DATA_DIRECTORY = 'ERR_LAUNCHER_INVALID_USER_DATA_DIRECTORY',
+  ERR_LAUNCHER_UNSUPPORTED_PLATFORM = 'ERR_LAUNCHER_UNSUPPORTED_PLATFORM',
+  ERR_LAUNCHER_NOT_INSTALLED = 'ERR_LAUNCHER_NOT_INSTALLED',
+}
+
+export function defaults<T>(val: T|undefined, def: T): T {
   return typeof val === 'undefined' ? def : val;
 }
 
 export async function delay(time: number) {
   return new Promise(resolve => setTimeout(resolve, time));
 }
 
+export class LauncherError extends Error {
+  public message: string;
+  public code: string;
+  constructor(message: string = 'Unexpected error', code?: string) {
```

Comment:
my previous comment about using


constructor(public message: string = '...', public code?: string) {
}

still stands

Reply:
@samccone Fixed in c6f69c5.